internal/controller/isbservicerollout_controller.go

```diff
@@ -330,7 +339,11 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 }
 
 func (r *ISBServiceRolloutReconciler) updateISBService(ctx context.Context, isbServiceRollout *apiv1.ISBServiceRollout, newISBServiceDef *kubernetes.GenericObject) error {
-	err := kubernetes.UpdateCR(ctx, r.restConfig, newISBServiceDef, "interstepbufferservices")
+	newISBServiceDefUns, err := kubernetes.ObjectToUnstructured(newISBServiceDef)
+	if err != nil {
+		return err
+	}
+	err = r.client.Update(ctx, newISBServiceDefUns)
```

Comment:
this was a good finding that it seems like we can maybe just use `r.client` across the board now - maybe our functions in `unstructured_util.go` will get replaced. Perhaps, the one thing we could do over there is have convenience functions that can take `GenericObject` so that we could reduce the code above for example to a single call?